javascript/26-Remove-Duplicates-from-Sorted-Array.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Linear 
+ * Time O(N) | Space O(1)
+ * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
+ * @param {number[]} nums
+ * @return {number}
+ */
+var removeDuplicates = function(nums) {
+
+    let left = 0;
+    let right = 1;
+    let delCount = 0;
+
+    while (right <= nums.length) {
+        if (nums[left] === nums[right]) {
+            right++;
+            delCount++;
+        } else {
+            nums.splice(left + 1, delCount);
```

Comment:
This makes it N^2

Reply:
Because of recording the index of each element? for example, [1,2,3,3,4,5] if we remove 3 from the 3rd index then all the elements' index will be updated after index 3. Is this the reason? If so can you suggest a better approach for removing elements?

Reply:
```js

/**
 * Two Pointer
 * Time O(N) | Space O(1)
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let [ left, right ] = [ 0, 0 ];
    
    while (right < nums.length) {/* Time O(N) */
        const [ leftVal, rightVal ] = [ nums[left], nums[right] ];
        const isEqual = (rightVal === leftVal);
        
        if (!isEqual) {
            left++;
            nums[left] = rightVal;
        }
        
        right++;
    }
    
    return (left + 1);
};

```

Reply:
That is really good. I updated the code please check.